src/allmydata/webish.py

```diff
@@ -1,43 +1,50 @@
 import re, time
+
+from functools import (
+    partial,
+)
+from cgi import (
+    FieldStorage,
+)
+
 from twisted.application import service, strports, internet
-from twisted.web import http, static
+from twisted.web import static
+from twisted.web.http import (
+    parse_qs,
+)
+from twisted.web.server import (
+    Request,
+    Site,
+)
 from twisted.internet import defer
 from twisted.internet.address import (
     IPv4Address,
     IPv6Address,
 )
-from nevow import appserver, inevow
 from allmydata.util import log, fileutil
 
 from allmydata.web import introweb, root
-from allmydata.web.common import MyExceptionHandler
 from allmydata.web.operations import OphandleTable
 
 from .web.storage_plugins import (
     StoragePlugins,
 )
 
-# we must override twisted.web.http.Request.requestReceived with a version
-# that doesn't use cgi.parse_multipart() . Since we actually use Nevow, we
-# override the nevow-specific subclass, nevow.appserver.NevowRequest . This
-# is an exact copy of twisted.web.http.Request (from SVN HEAD on 10-Aug-2007)
-# that modifies the way form arguments are parsed. Note that this sort of
-# surgery may induce a dependency upon a particular version of twisted.web
-
-parse_qs = http.parse_qs
-class MyRequest(appserver.NevowRequest, object):
+class TahoeLAFSRequest(Request, object):
     fields = None
     _tahoe_request_had_error = None
```

Comment:
I could not find any actual use for `_tahoe_request_had_error`. Could this be removed?